nautobot/utilities/utils.py

```diff
@@ -869,7 +869,9 @@ def get_filterset_parameter_form_field(model, parameter):
     form_field = field.field
 
     # TODO(Culver): We are having to replace some widgets here because multivalue_field_factory that generates these isn't smart enough
-    if isinstance(field, NumberFilter):
+    if isinstance(field, CustomFieldFilterMixin):
+        form_field = field.custom_field.to_form_field()
```

Comment:
Sadly long-term, we should merge this logic here (in `get_filterset_parameter_form_field`) with `multivalue_field_factory` and the field+filter+widget association.